django_mongodb/query.py

```diff
@@ -0,0 +1,359 @@
+        child_queries = list(
+            filter(bool, [child.get_mongo_search(compiler, connection) for child in self.children])
+        )
+        if len(child_queries) == 0:
+            return {}
+        elif self.connector == "AND":
+            return {"compound": {"must": child_queries}}
+        elif self.connector == "OR":
+            return {"compound": {"should": child_queries}}
+        else:
+            raise Exception(f"Unsupported connector: {self.connector}")
+
+
+class MongoColSelect:
+    def __init__(self, col: Col, alias: str | None, mongo_meta):
+        self.col = col
+        self.mongo_meta = mongo_meta
+        self.alias = alias
+
+    def get_mongo(self):
+        return {"$project": {(self.alias or self.col.target.attname): f"${self.col.target.column}"}}
+
+
+class MongoValueSelect:
+    def __init__(self, col: Value, alias: str | None, mongo_meta):
+        self.col = col
+        self.mongo_meta = mongo_meta
+        self.alias = alias
+
+    def get_mongo(self):
+        return {"$project": {(self.alias): self.col.value}}
+
+
+class MongoCountSelect:
+    def __init__(self, col: Count, alias: str | None, mongo_meta):
+        self.col = col
+        self.mongo_meta = mongo_meta
+        self.alias = alias
+
+    def get_mongo(self):
+        return {
+            "$group": {"_id": None, "_count": {"$sum": 1}},
+            "$project": {"_id": None, (self.alias or self.col.output_field.column): "$_count"},
+        }
+
+
+class MongoSelect:
+    def __init__(self, _cols: list[tuple[Expression, tuple, str | None]], mongo_meta):
+        self.mongo_meta = mongo_meta
+        self.cols = []
+        for col in _cols:
+            [column, _, alias] = col
+            match column:
+                case Col():
+                    self.cols.append(MongoColSelect(column, alias, mongo_meta))
+                case Value():
+                    self.cols.append(MongoValueSelect(column, alias, mongo_meta))
+                case Count():
+                    self.cols.append(MongoCountSelect(column, alias, mongo_meta))
+                case SearchVector():
+                    pass  # ignoring search vector in results
+                case _:
+                    raise NotImplementedError(f"Select expression not implemented: {col}")
+
+    def get_mongo(self):
+        pipeline_dict: dict[str, dict] = OrderedDict()
+        pipeline_dict["$group"] = dict()
+        pipeline_dict["$project"] = dict()
+        for col in self.cols:
+            mongo_query = col.get_mongo()
+            for key, item in mongo_query.items():
+                pipeline_dict[key].update(item)
+
+        return [{key: item} for key, item in pipeline_dict.items() if item]
+
+
+class MongoOrdering:
+    """MongoDB Query Node for Ordering"""
+
+    def __init__(self, query: Query):
+        self.query = query
+        self.order = query.order_by
+
+    def get_mongo_order(self, attname_as_key=False):
```

Comment:
in aggregation queries, we already have the document with Django attribute names as keys and are setting `attname_as_key=True`